### DISCUSSION TITLE
IP address permissions

### DISCUSSION THREAD

**Discussion by andershagman** [netbox-community/netbox#6504] [Q&A]
Hi

I want to restrict permissions for certen groups to add, modify and delete ip addresses in specific prefixes with a certen role.
But there is no prefix__role or parent__role in ipam -> address.

How can I do this?

**Comment by dw72** [comment]
I have the same problem. Can anyone give a hint on how to do this?

For prefix /24 I used _address__startswith_ but this workaround will not work in all cases.

**Reply by candlerb** [reply]
IP Address and Prefix have no explicit linkage in the data model.

To find the "parent" prefix, you'd have to either:
1. do a database query which picks all prefixes that enclose the IP, sorted by prefix length, and take the first; or
2. make use of the prefix length (zero off all the bits past the prefix, and search for a prefix that matches this exactly)

However, I've looked in the [model](https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/models/ip.py#L525-L733) and I can't find any property which does either of those.

In the GUI when you view an IP address you can see its parents, but that logic is in the [view](https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/views.py#L522-L530), not the model.

**Reply by andershagman** [reply]
One way to go is to create a tenant group "Group 1 and 2" and two tenants "group 1" and "group 2". Where group 1 is the masters and group 2 is the the people with less add, change and delete rights.

Then create three separate permissions. 
1. view ip-addresses, prefixes and vrf -> if tenant_group_name = "Group 1 and 2"
2. view tenant -> if tenant_name = "group 2"
3. add, change, delete adresses -> if tenant_name = "group 2"

Add those permissions to group 2.

Put all desired prefixes in tenant "group 2"
Put all forbidden addresses in those prefixes in tenant "group 1". Group 2 can see them but not change them (eg gateway adress).

Group 2 can now add, change and delete addresses as long as they add the right tenant to a new address. They can only choose from one tenant.
